### ISSUE TITLE
Formatting of certain brace types is not configurable.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#60390]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Formatting-of-certain-brace-types-is-not/1702670)._

---
[severity:It's more difficult to complete my work]
While we can format curly braces (`{`, and `}`), we don't currently have the same level of settings for other brace types, such as round braces (`(` and `)`), or square braces (`[`, and `]`).

This makes using CSOM (C# Object Notation), and _other declarative_ formatting styles hard to use, because the auto-formatter just completely mangles them, and there are no options to allow us control over this type of formatting.

**Examples:**
```
var x = new SomeObject
{
    Property1 = MyMethodWithTooManyParameters
    (
        param1,
        string. Join(", ", param2). Where(x => x is not null),
        AnotherMethod
        (
            param3,
            param4
        )
    ),
    Property2 = "Foo",
    Property3 = 12
};
```
```
public
(
    IEnumerable<ObjectA> ResultsA,
    IEnumerable<ObjectB> ResultsB
)
GetSomeObjects(ObjectC inputC)
{
    // ...
}
```

**In Closing:**
Please give us control over the various styles of braces and newlines to the same degree that we have for curly braces.

---
#### Feedback Bot on 3/24/2022, 11:01 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by cremor** [comment]
Yes, this would be very helpful. But opening and closing braces need to have separate options.

For example, a "wrap every argument"-"indent all arguments" code fix of the code

```cs
MyMethodWithTooManyParameters(param1, param2, param3, param4);
```

currently produces

```cs
MyMethodWithTooManyParameters(
    param1,
    param2,
    param3,
    param4);
```

and should then also have an option to produce

```cs
MyMethodWithTooManyParameters(
    param1,
    param2,
    param3,
    param4
);
```

**Comment by allisonchou** [comment]
Design meeting notes (4/4/2022):
- Retrieve additional data on formatting (e.g. what customers want, how often current formatting options are used, etc.)
- Create formatting working group/design team to discuss these formatting issues further

**Comment by CyrusNajmabadi** [comment]
Tracking with https://github.com/dotnet/roslyn/issues/31691
